lectures/sympy.md

```diff
@@ -0,0 +1,992 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.4
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+(sympy)=
+```{raw} jupyter
+<div id="qe-notebook-header" align="right" style="text-align:right;">
+        <a href="https://quantecon.org/" title="quantecon.org">
+                <img style="width:250px;display:inline;" width="250px" src="https://assets.quantecon.org/img/qe-menubar-logo.svg" alt="QuantEcon">
+        </a>
+</div>
+```
+
+# {index}`SymPy <single: SymPy>`
+
+```{index} single: Python; SymPy
+```
+
+```{contents} Contents
+:depth: 2
+```
+
+## Overview
+
+Unlike numerical libraries that deal with values, [SymPy](https://www.sympy.org/en/index.html) focuses on manipulating mathematical symbols and expressions directly.
+
+SymPy provides [a wide range of features](https://www.sympy.org/en/features.html) including 
+
+- Symbolic Expression
+- Equation Solving
+- Simplification
+- Calculus
+- Matrices
+- Discrete Math, etc.
+
+These functions make SymPy a popular open-source alternative to other proprietary symbolic computation software such as Mathematica.
+
+In this lecture, we will explore some of the functionalities of SymPy and demonstrate how to use basic SymPy functions to solve economic models.
+
+## Getting Started
+
+Let’s first import the library and initialize the printer for symbolic output
+
+```{code-cell} ipython3
+from sympy import *
+from sympy.plotting import plot, plot3d_parametric_line, plot3d
+from sympy.solvers.inequalities import reduce_rational_inequalities
+from sympy.stats import Poisson, Exponential, Binomial, density, moment, E, cdf
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+# Enable the mathjax printer
+init_printing(use_latex='mathjax')
+```
+
+## Symbolic algebra
+
+### Symbols
+
+Before we start manipulating the symbols, we initialize some symbols to work with
+
+```{code-cell} ipython3
+x, y, z = symbols('x y z')
+```
+
+Symbols are the basic units for symbolic computation in SymPy.
+
+### Expressions
+
+We can now use symbols `x`, `y`, and `z` to build expressions and equations.
+
+Here we build a simple expression first
+
+```{code-cell} ipython3
+expr = (x+y) ** 2
+expr
+```
+
+We can expand this expression with the `expand` function
+
+```{code-cell} ipython3
+expand_expr = expand(expr)
+expand_expr
+```
+
+and factorize it back to the factored form with the `factor` function
+
+```{code-cell} ipython3
+factor(expand_expr)
+```
+
+We can solve this expression
+
+```{code-cell} ipython3
+solve(expr)
+```
+
+Note this is equivalent to solving the following equation for `x`
+
+$$
+(x + y)^2 = 0 
+$$
+
+Another way to solve this equation is to use `solveset`.
+
+`solveset` allows users to define the domain of `x` in the evaluation and output a set as a solution
+
+```{code-cell} ipython3
+solveset(expr, x, domain=S.Naturals)
+```
```

Comment:
to here in the interest of brevity --- we don't have to mention all possibilities.